GUI/Main.cs

```diff
@@ -555,18 +555,22 @@ private void CurrentInstanceUpdated(bool onStartup)
 
             bool repoUpdateNeeded = configuration.RefreshOnStartup
                 || !RegistryManager.Instance(CurrentInstance).registry.HasAnyAvailable();
-            if (onStartup && repoUpdateNeeded)
+            if (allowRepoUpdate && repoUpdateNeeded)
             {
                 UpdateRepo();
+
+                // Update the filters AFTER UpdateRepo() completed.
+                // Since this happens with a backgroundworker, Filter() is added as callback for RunWorkerCompleted.
+                void filterUpdate (object sender, RunWorkerCompletedEventArgs e) { Filter((GUIModFilter)configuration.ActiveFilter); }
+                m_UpdateRepoWorker.RunWorkerCompleted += filterUpdate;
```

Comment:
Won't this add the handler every time this block runs? If we do this multiple times, I think you'd have redundant actions when the worker completes.

Reply:
That's... uhhhm... unfortunate.
Let me try to move the filter update before `UpdateRepo()`, see if that works too.

Reply:
Okay I just discovered that we only save the current filter `OnFormClosing()`.
It makes sense saving it before each instance change, doesn't it?

Reply:
Another option is to remove the handler after it runs with `-=`, or just add it once globally somewhere.

Reply:
I don't want to add it globally, since it is designed to explicitly run when the instance changed (by loading the old filter from the config),
which, if I wouldn't have added that the new filter is saved with each filter change, would "reset" the filter again to what it was at CKAN startup with each `UpdateRepo()` call, so even after refreshes and mod installs and so on.

So thanks to the config save now I _could_ add it globally, but it doesn't fit there IMO.
I'm going with removing it again.